spec/classes/init_spec.rb

```diff
@@ -1,91 +1,106 @@
+          client_package: 'custom-gluster-client',
+          version: '3.1.4',
+          repo: false,
+        }
+        end
+        it 'should create gluster::install' do
+          should create_class('gluster::install').with(
+            server: true,
+            server_package: 'custom-gluster-server',
+            client: true,
+            client_package: 'custom-gluster-client',
+            version: '3.1.4',
+            repo: false,
+          )
+        end
+        it 'should manage the Gluster service' do
+          should create_class('gluster::service').with(
+            ensure: true,
+          )
+        end
+        it 'should install custom-gluster-client and custom-gluster-server' do
+          should create_package('custom-gluster-client')
+          should create_package('custom-gluster-server')
+        end
       end
-    end
 
-    context 'when volumes defined' do
-      let :params do
-        { :volumes =>
+      context 'when volumes defined' do
+        let :facts do
+          super().merge(
+            gluster_binary: '/sbin/gluster',
+            gluster_peer_list: 'example1,example2',
+            gluster_volume_list: 'gl1.example.com:/glusterfs/backup,gl2.example.com:/glusterfs/backup'
+          )
+        end
+        let :params do
           {
-            'data1' => {
-              'replica' => 2,
-              'bricks'  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
-              'options' => ['server.allow-insecure: on'],
+            volumes:
+            {
+              'data1' => {
+                'replica' => 2,
+                'bricks'  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
+                'options' => ['server.allow-insecure: on'],
+              }
             }
           }
-        }
-      end
-      it 'should create gluster::volume' do
-        should contain_gluster__volume('data1').with(
-          :name => 'data1',
-          :replica => 2,
-          :bricks  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
-          :options => ['server.allow-insecure: on'],
-        )
+        end
+        it 'should create gluster::volume' do
+          should contain_gluster__volume('data1').with(
+            name: 'data1',
+            replica: 2,
+            bricks: ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
+            options: ['server.allow-insecure: on'],
+          )
+        end
       end
-    end
 
-    context 'when volumes incorrectly defined' do
-      let :params do
-        { :volumes =>
+      context 'when volumes incorrectly defined' do
+        let :params do
           {
-            'data1' => %w(this is an array)
+            volumes: { 'data1' => %w(this is an array) }
           }
-        }
-      end
-      it 'should fail' do
-        expect {
-          should contain_gluster__volume('data1')
-        }.to raise_error(Puppet::Error, //)
+        end
+        it 'should fail' do
```

Comment:
according to @jyaworski this titles will soon change with rspec-rubocop.

we can preemptively change it to `it 'fails'`